holidays/financial/national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,170 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomHinduHolidays, _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG
+from holidays.groups import ChristianHolidays, HinduCalendarHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SAT_TO_NONE, SUN_TO_NONE
+
+
+class NationalStockExchangeOfIndia(
+    ObservedHolidayBase, HinduCalendarHolidays, ChristianHolidays, IslamicHolidays
+):
+    """National Stock Exchange of India (NSE) holidays.
+
+    References:
+        * <https://web.archive.org/web/20250821175252/https://www.nseindia.com/resources/exchange-communication-circulars>
+
+    Historical data:
+        * [2022](https://web.archive.org/web/20250821071611/https://nsearchives.nseindia.com/content/circulars/CMTR50560.pdf)
+        * [2023](https://web.archive.org/web/20250821071635/https://nsearchives.nseindia.com/content/circulars/CMTR54757.pdf)
+        * [2024](https://web.archive.org/web/20250821071650/https://nsearchives.nseindia.com/content/circulars/CMTR59722.pdf)
+        * [2025](https://web.archive.org/web/20250624132016/https://nsearchives.nseindia.com/content/circulars/CMTR65587.pdf)
+    """
+
+    market = "XNSE"
+    default_language = "en_IN"
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    supported_languages = ("en_IN", "en_US", "hi")
+    # NSE launched its services in 1994.
+    start_year = 1994
```

Comment:
We need full year here
```suggestion
    # NSE launched its services in 1994.
    start_year = 1995
```